Templates/FaceLocator-NodeJS/index.js

```diff
@@ -0,0 +1,58 @@
+// Setup
+// 1) Go to https://www.microsoft.com/cognitive-services/en-us/computer-vision-api 
+//    Sign up for computer vision api
+// 2) Go to Function app settings -> App Service settings -> Settings -> Application settings
+//    create a new app setting Vision_API_Subscription_Key and use Computer vision key as value
+// 3) Go to Function app settings -> App Service settings -> Tools -> Console
+//    Enter the following commands: 
+//    > cd <functionName>
+//    > npm install
+module.exports = function (context, image) {
+    context.log("Image Size:", image.length);
+
+    var request = require('request-promise');
+
+    var options = {
+        uri: "https://api.projectoxford.ai/vision/v1.0/analyze?visualFeatures=Faces",
+        method: 'POST',
+        body: image,
+        headers: {
+            'Content-Type': 'application/octet-stream',
+            'Ocp-Apim-Subscription-Key': process.env.Vision_API_Subscription_Key
+        }
+    };
+
+    request(options)
+        .then(function (response) {
+            response = JSON.parse(response);
+
+            if (typeof response == 'undefined' && typeof response.faces.length === 0) {
+                context.done();
+                return;
+            }
+            context.log("From Vision Api:", response);
+
+            var guid = require('node-uuid');
+            var faces = [];
```

Comment:
I'd like a more functional style, for loops are meh.  The object.assign is questionable, but it's shorter.

```js
var faces = response.faces.map((face) => {
  var faceMapping = {
    rowKey: ...,
    imageFile: ...,
    partitionKey: ...
  }
  Object.assign(faceMapping, face.faceRectangle);
  return faceMapping;
});